bridge/sam/pool.go

```diff
@@ -0,0 +1,399 @@
+package sam
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type pool struct {
+	// write-lock is called only when changing validators process
+	// otherwise read-lock is called
+	// Changing validators will occur the most rarely (once per epoch)
+	changeValidatorsLock sync.RWMutex
+	validators           []types.Address
+	threshold            uint64 // required number of signatures for ready
```

Comment:
Please consider renaming this to `signatureThreshold`, as it's not immediately clear from the usage what it represents